TrackerMC/src/StrawDigiBundle.cc

```diff
@@ -0,0 +1,61 @@
+// Bundle together references to various StrawDigi products, to circumvent
+// triplicated-loops which pop up when iterating over digis
+//
+// Ed Callaghan, 2023
+
+#include <Offline/DataProducts/inc/TrkTypes.hh>
+#include <Offline/TrackerMC/inc/StrawDigiBundle.hh>
+
+namespace mu2e{
+  StrawDigiMC empty_mc;
+}
+
+namespace mu2e{
+  StrawDigiBundle::StrawDigiBundle(const StrawDigi digi,
+                                   const StrawDigiADCWaveform adcs,
+                                   const StrawDigiMC mc):
+                                    digi(digi),
+                                    adcs(adcs),
+                                      mc(mc){
+    /**/
+  }
+
+  StrawDigiBundle::StrawDigiBundle(const StrawDigi digi,
+                                   const StrawDigiADCWaveform adcs):
+                                    digi(digi),
+                                    adcs(adcs),
+                                      mc(empty_mc){
+    /**/
+  }
+
+  StrawDigiBundle::StrawDigiBundle(const StrawDigiBundle& bundle):
+                                    digi(bundle.GetStrawDigi()),
+                                    adcs(bundle.GetStrawDigiADCWaveform()),
+                                      mc(bundle.GetStrawDigiMC()){
+    /**/
+  }
+
+  const StrawDigi StrawDigiBundle::GetStrawDigi() const{
+    const StrawDigi rv = this->digi;
```

Comment:
I think you only need the "this->" construct in rare circumstances, here you can just do 
return digi;

Reply:
I've removed all instances of "this->" when accessing data members, but left it in place for calling class methods.

Reply:
I personally find this hard to look at, but I see it does occur elsewhere in the code, so I won't object.